### PR TITLE
feat: Populate provided VPC with subnets so that a VPC can be looked up

### LINKED ISSUE #13: feat: Add Construct to create subnets in pre-provisioned VPC
As a developer that:
- Doesn't have IAM permissions to create a VPC
- Must have a VPC created for them by another entity (e.g. cloud center of excellence, security team)
- Are provided with:
  - VPC
  - Private route table (route to internal enterprise)
  - Isolated route table  (route to local VPC)

I want a construct for creating subnets inside that VPC with appropriate tags so that it can be [imported](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2-readme.html#importing-an-existing-vpc) using [`fromLookup()`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#static-fromwbrlookupscope-id-options) or [fromVpcAttributes](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#static-fromwbrvpcwbrattributesscope-id-attrs).

So that an enterprise provided VPC created outside of the CDK project can be used as an [IVpc](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.IVpc.html)

### PR DESCRIPTION
**Pull request by maafk**
Fixes #13 

Adds two constructs for populating a VPC provided by an enterprise.

Scenario is that the developer doesn't have permissions to create their own VPC, but are provided a VPC ID, and two route tables. One route table with routes to the enterprise network, and one route table for routing locally within the VPC.

`PopulateWithConfig` allows a user to pass an array of `SubnetConfig`s

`SplitVpcEvenly` allows a user to split the subnet evenly, using sensible defaults, but also providing options to customize

### CODE REVIEW

File: src/constructs/populateProvidedVpc.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    this._vpcId = props.vpcId;
+    this._vpcCidr = props.vpcCidr;
+    this._routeTableId = props.routeTableId;
+    this._cidrBits = props.cidrBits || '6';
+    this._numberOfAzs = props.numberOfAzs || 3;
+    this._subnetTag = props.subnetTag || SubnetTag.PRIVATE;
+
+    // based on number of az, create an array of az strings
+    const azs: string[] = [];
+    for (let i = 0; i < this._numberOfAzs; i++) {
```
[the remark this answers is not in the document]
Comment (maafk) [reply src/constructs/populateProvidedVpc.ts:245]: Added in a check that throws an error if invalid number of AZs provided

File: src/constructs/populateProvidedVpc.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+   * @default 3
+   */
+  readonly numberOfAzs?: number;
+  /**
+   * @default SubnetTag.PRIVATE
+   */
+  readonly subnetTag?: SubnetTag;
+}
+/**
+ * Splits a VPC evenly between 3 AZs, and attaches a provided route table to each, and labels
```
[the remark this answers is not in the document]
Comment (maafk) [reply src/constructs/populateProvidedVpc.ts]: Updated

File: src/constructs/populateProvidedVpc/populateProvidedVpc.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  PRIVATE = 'Private',
+  /**
+   * {"Name": "aws-cdk:subnet-type", "Value": "Public"} tag on a subnet
+   */
+  PUBLIC = 'Public',
+  /**
+   * {"Name": "aws-cdk:subnet-type", "Value": "Isolated"} tag on a subnet
+   */
+  ISOLATED = 'Isolated',
+}
```
Comment (joel-aws) [inline src/constructs/populateProvidedVpc/populateProvidedVpc.ts:25]: CDK v2 uses [different enums](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.SubnetType.html) for SubnetTypes. Recommend:
* Verifying what tags the new enums create in VPCs.
* Using the same enums here.

**Review by joel-aws — APPROVED** [review]
LGTM!

**Pull request merged**